ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineExchangeTransitionConfiguration.java

```diff
@@ -102,4 +113,29 @@ private JsonRpcResponse respondWith(
       final EngineExchangeTransitionConfigurationResult transitionConfiguration) {
     return new JsonRpcSuccessResponse(requestId, transitionConfiguration);
   }
+
+  // Handler is a member func in order to refer to our vertx instance,
+  // qosTimeout is a parameter for testing
+  Handler<Long> qosHandler(final long qosTimeout) {
+    return z -> {
+      if (getLastCallMillis() + qosTimeout < System.currentTimeMillis()) {
+        logQosFailure(qosTimeout);
+      }
+      resetQosHandler(qosTimeout, qosHandler(qosTimeout));
+    };
+  }
+
+  void resetQosHandler(final long qosTimeout, final Handler<Long> qosHandler) {
+    Optional.ofNullable(qosTimerId.get()).ifPresent(timerVertx::cancelTimer);
+    qosLastCall.set(System.currentTimeMillis());
+    qosTimerId.set(timerVertx.setTimer(qosTimeout, qosHandler));
+  }
+
+  void logQosFailure(final long qosTimeout) {
+    LOG.warn("not called in {} seconds, consensus client may not be connected", qosTimeout / 1000L);
+  }
+
+  long getLastCallMillis() {
+    return Optional.ofNullable(qosLastCall.get()).orElse(0L);
```

Comment:
Here we have the same behavior as above, qosLastCall.get() cannot be null.

Reply:
That config should work, I used the same mergemock config to test :).  Did you wait 2 minutes to see the warning?

```
2022-03-11 04:02:47.029-08:00 | main | INFO  | Runner | Starting Ethereum main loop ... 
..
2022-03-11 04:04:42.218-08:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers. Current peers: 0
2022-03-11 04:04:46.911-08:00 | vert.x-eventloop-thread-0 | WARN  | EngineExchangeTransitionConfiguration | not called in 120 seconds, consensus client may not be connected
...
2022-03-11 04:06:42.388-08:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers. Current peers: 0
2022-03-11 04:06:46.919-08:00 | vert.x-eventloop-thread-0 | WARN  | EngineExchangeTransitionConfiguration | not called in 120 seconds, consensus client may not be connected
...
2022-03-11 04:10:42.723-08:00 | EthScheduler-Timer-0 | INFO  | FullSyncTargetManager | No sync target, waiting for peers. Current peers: 0
2022-03-11 04:10:46.922-08:00 | vert.x-eventloop-thread-0 | WARN  | EngineExchangeTransitionConfiguration | not called in 120 seconds, consensus client may not be connected
```